core/field_checkbox.js

```diff
@@ -42,15 +42,28 @@ goog.require('Blockly.utils.Size');
  *    changes to the field's value. Takes in a value ('TRUE' or 'FALSE') &
  *    returns a validated value ('TRUE' or 'FALSE'), or null to abort the
  *    change.
+ * @param {Object=} opt_config A map of options used to configure the field.
+ *    See the documentation for a list of properties this parameter supports.
+ *    https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/checkbox#creation
  * @extends {Blockly.Field}
  * @constructor
  */
-Blockly.FieldCheckbox = function(opt_value, opt_validator) {
+Blockly.FieldCheckbox = function(opt_value, opt_validator, opt_config) {
+  /** @type {?string} @private */
+  Blockly.FieldCheckbox.prototype.checkChar_ = null;
```

Comment:
Let's move this to the bottom of the constructor (especially after calling the superclass constructor).
and defined the type with multi line comments like so: 
```

/**
 * Character for the check mark. Used to apply a different check mark
 * character to individual fields.
 * @type {?string}
 * @private
 */
```

Reply:
Also, when it's in the constructor. It's simply: 
``this.checkChar_ = null;``

Reply:
> Also, when it's in the constructor. It's simply:
> `this.checkChar_ = null;`

Derp. Fixed.

> Let's move this to the bottom of the constructor

I don't think it can be moved to the bottom b/c then it would override the configured option.

Option 1:
We could move it to the bottom and then call a config function after it. Problem is some fields (like the number field) need to be configured before the super-constructor is called so that the class validator works properly.

Option 2:
Remove the intial `this.checkChar_ = null; ` line and instead document the `this.checkChar_ = opt_config['checkCharacter']` line.

Option 3:
Leave as is.

Reply:
My recommendation would be to do: 
- super
- initialization
- config

I recognize that in some cases (number field), you might have to do initialization first, and that's okay as an exception to the rule.

Reply:
> My recommendation would be to do:
> 
> * super
> * initialization
> * config

Ok, can I fix as a separate PR since this affects all fields?

Reply:
Yep